dvc/cache/base.py

```diff
@@ -215,9 +219,13 @@ def _transfer_directory(
         dir_info = DirInfo()
 
         with Tqdm(total=1, unit="Files", disable=no_progress_bar) as pbar:
-            for entry_info, entry_hash in self._transfer_directory_contents(
+            for (
+                entry_info,
+                (entry_tmp_info, entry_hash),
+            ) in self._transfer_directory_contents(
                 from_tree, from_info, jobs, pbar
             ):
+                self.add(entry_tmp_info, self.tree, entry_hash)
```

Comment:
Now we are checking if the file is already in cache before doing a final move, so need to not forget to clean it up.